src/js/program/program-controller.js

```diff
@@ -58,6 +56,7 @@ export default class ProgramController extends Eventable {
                 this._setActiveMedia(mediaController);
                 // Initialize the provider last so it's setting properties on the (newly) active media model
                 mediaController.provider.init(item);
+                model.setActiveItem(index);
```

Comment:
Can this come after `this.providerPromise` ? If that doesn't work, the one on line 84 won't work either.

Reply:
It should work if play() always follows `this.providerPromise`.

Reply:
Just after `this._setActiveMedia` is good enough.